services/wallet/router/router_v2.go

```diff
@@ -38,8 +38,10 @@ type RouteInputParams struct {
 	AddrFrom             common.Address          `json:"addrFrom" validate:"required"`
 	AddrTo               common.Address          `json:"addrTo" validate:"required"`
 	AmountIn             *hexutil.Big            `json:"amountIn" validate:"required"`
+	AmountOut            *hexutil.Big            `json:"amountOut"`
 	TokenID              string                  `json:"tokenID" validate:"required"`
 	ToTokenID            string                  `json:"toTokenID"`
+	SlippagePercentage   float32                 `json:"slippage"`
```

Comment:
Hmm this struct would be part of the public API right? I think we try not to use floats for user-entered amounts do to loss of precision. I was thinking of using the Basis Points representation in the API as well, since we don't allow entering more than 2 decimals anyway.

Reply:
Yes, it's part of the API. 

Well, we can change it to string and you can provide float as a string. Why I set it to %, but not to basis points, cause we can use the same prop for other types that are expecting slippage in percentage. 
Also I wouldn't say that matter in this context cause the loss of precision you're pointing to usually comes up with higher precision, like if we were to use 7, 8, 10+ decimal places, but for two it doesn't matter in this context. 

I mean what's the difference in terms of slippage if the user sets it to 
`2.3`%, but statusgo receives it like `2.29999995231628417969`, not a big deal in this context.

Reply:
> Well, we can change it to string and you can provide float as a string. Why I set it to %, but not to basis points, cause we can use the same prop for other types that are expecting slippage in percentage.

I don't think that will make a difference since we're converting to float anyway.

> Also I wouldn't say that matter in this context cause the loss of precision you're pointing to usually comes up with higher precision, like if we were to use 7, 8, 10+ decimal places, but for two it doesn't matter in this context.

Yeah, for sure it doesn't matter for whatever calculations need to be performed w.r.t. slippage. I guess it's more of a general discussion about handling user input than whatever real implications it will have for this particular case.